nucypher-core/src/dkg.rs

```diff
@@ -21,6 +109,280 @@ pub enum FerveoVariant {
 }
 
+pub mod request_keys {
+    use alloc::boxed::Box;
+    use alloc::string::String;
+    use core::fmt;
+    use generic_array::typenum::Unsigned;
+    use generic_array::{typenum::U32, ArrayLength, GenericArray};
+    use hkdf::Hkdf;
+    use rand::SeedableRng;
+    use rand_chacha::ChaCha20Rng;
+    use rand_core::{CryptoRng, OsRng, RngCore};
+    use serde::{Deserialize, Serialize};
+    use sha2::Sha256;
+    use x25519_dalek::{PublicKey, SharedSecret, StaticSecret};
+
+    use crate::secret_box::SecretBox;
+    use zeroize::ZeroizeOnDrop;
+
+    use crate::versioning::{
+        messagepack_deserialize, messagepack_serialize, ProtocolObject, ProtocolObjectInner,
+    };
+
+    /// A Diffie-Hellman shared secret
+    #[derive(ZeroizeOnDrop)]
+    pub struct RequestSharedSecret(pub(crate) SharedSecret);
+
+    /// Implementation of Diffie-Hellman shared secret
+    impl RequestSharedSecret {
+        /// Create new shared secret from underlying library.
+        pub fn new(shared_secret: SharedSecret) -> Self {
+            Self(shared_secret)
+        }
+
+        /// Convert this shared secret to a byte array.
+        #[inline]
+        pub fn to_bytes(&self) -> [u8; 32] {
+            self.0.to_bytes()
+        }
+
+        /// View this shared secret key as a byte array.
+        #[inline]
+        pub fn as_bytes(&self) -> &[u8; 32] {
+            self.0.as_bytes()
+        }
+    }
+
+    impl AsRef<[u8]> for RequestSharedSecret {
+        /// View this shared secret key as a byte array.
+        #[inline]
+        fn as_ref(&self) -> &[u8] {
+            self.0.as_bytes()
+        }
+    }
+
+    impl fmt::Display for RequestSharedSecret {
+        /// Format shared secret information.
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            write!(f, "RequestSharedSecret...")
+        }
+    }
+
+    /// A request public key.
+    #[derive(PartialEq, Eq, Hash, Copy, Clone, Debug, Serialize, Deserialize)]
+    pub struct RequestPublicKey(pub(crate) PublicKey);
+
+    /// Implementation of request public key
+    impl RequestPublicKey {
+        /// Convert this public key to a byte array.
+        #[inline]
+        pub fn to_bytes(&self) -> [u8; 32] {
+            self.0.to_bytes()
+        }
+
+        /// View this public key as a byte array.
+        #[inline]
+        pub fn as_bytes(&self) -> &[u8; 32] {
+            self.0.as_bytes()
+        }
+    }
+
+    impl AsRef<[u8]> for RequestPublicKey {
+        /// View this public key as a byte array.
+        #[inline]
+        fn as_ref(&self) -> &[u8] {
+            self.0.as_bytes()
+        }
+    }
+
+    impl fmt::Display for RequestPublicKey {
+        /// Format public key information.
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            let bytes = self.as_ref();
+            let bytes = if bytes.len() > 8 { &bytes[..8] } else { bytes };
+            write!(f, "RequestPublicKey: {}", hex::encode(bytes),)
```

Comment:
I think it's a typo:
```suggestion
            write!(f, "RequestPublicKey: {}", hex::encode(bytes))
```
`write!` takes an arbitrary number of args, so it wouldn't be caught by the type checker.